extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/media/GoogleMediaImporter.java

```diff
@@ -155,6 +156,7 @@ public ImportResult importItem(
       // Nothing to do
       return ImportResult.OK;
     }
+    final GPhotosUpload gPhotosUpload = new GPhotosUpload(jobId, idempotentImportExecutor, authData);
```

Comment:
one thing to call out (maybe make explicit in a comment) is that this hsould be created per request so that in the case where this might be multithreaded, we dont mix up creds (similar to how the interface is called using getorcreate and keeps a mapping of job id -> interfaces

just to prevent someone from coming along in the future and making this a class variable instead of a temp scoped one.

Reply:
good call! done